distributed/diagnostics/tests/test_worker_plugin.py

```diff
@@ -222,36 +223,30 @@ def teardown(self, worker):
             assert self._called
             return super().teardown(worker)
 
-    @gen_cluster(client=True, nthreads=[("", 1)])
-    async def test(c, s, a):
-
-        await c.register_worker_plugin(ReleaseKeyDeprecated())
-        fut = await c.submit(inc, 1, key="task")
-        assert fut == 2
+    await c.register_worker_plugin(ReleaseKeyDeprecated())
 
     with pytest.deprecated_call(
         match="The `WorkerPlugin.release_key` hook is depreacted"
     ):
-        test()
+        assert await c.submit(inc, 1, key="x") == 2
+        while "x" in a.tasks:
+            await asyncio.sleep(0.01)
 
 
-def test_assert_no_warning_no_overload():
+@gen_cluster(client=True, nthreads=[("", 1)])
+async def test_assert_no_warning_no_overload(c, s, a):
     """Assert we do not receive a deprecation warning if we do not overload any
     methods
     """
 
     class Dummy(WorkerPlugin):
         pass
 
-    @gen_cluster(client=True, nthreads=[("", 1)])
-    async def test(c, s, a):
-
-        await c.register_worker_plugin(Dummy())
-        fut = await c.submit(inc, 1, key="task")
-        assert fut == 2
-
     with pytest.warns(None):
-        test()
+        await c.register_worker_plugin(Dummy())
+        assert await c.submit(inc, 1, key="x") == 2
+        while "x" in a.tasks:
+            await asyncio.sleep(0.01)
```

Comment:
I'm curious why this addition was needed

Reply:
Because nothing tests for the existence of the release_key method in the plugin until a key is released